File: UsuariosAPI/src/main/java/com/clinica/api/personal_service/model/Doctor.java
package com.clinica.api.personal_service.model;

import jakarta.persistence.*;

@Entity
@Table(name = "doctor")
public class Doctor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_doctor")
    private Long id;

    @Column(name = "tarifa_consulta")
    private Integer tarifaConsulta;

    @Column
    private Long sueldo;

    @Column
    private Long bono;

    @Column(nullable = false)
    private Boolean activo = true;

    @OneToOne
    @JoinColumn(name = "id_usuario", nullable = false)
    private Usuario usuario;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_especialidad")
    private Especialidad especialidad;

    public Doctor() {
    }

    @PrePersist
    void ensureDefaults() {
        if (activo == null) {
            activo = true;
        }
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Integer getTarifaConsulta() {
        return tarifaConsulta;
    }

    public void setTarifaConsulta(Integer tarifaConsulta) {
        this.tarifaConsulta = tarifaConsulta;
    }

    public Long getSueldo() {
        return sueldo;
    }

    public void setSueldo(Long sueldo) {
        this.sueldo = sueldo;
    }

    public Long getBono() {
        return bono;
    }

    public void setBono(Long bono) {
        this.bono = bono;
    }

    public Usuario getUsuario() {
        return usuario;
    }

    public void setUsuario(Usuario usuario) {
        this.usuario = usuario;
    }

    public Especialidad getEspecialidad() {
        return especialidad;
    }

    public void setEspecialidad(Especialidad especialidad) {
        this.especialidad = especialidad;
    }

    public Boolean getActivo() {
        return activo;
    }

    public void setActivo(Boolean activo) {
        this.activo = activo;
    }
}

File: usuario-service/bin/main/com/clinica/api/usuario_service/model/Doctor.kt
package com.clinica.api.usuario_service.model

import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table(name = "doctor")
data class Doctor(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_doctor")
    var id: Long? = null,

    @Column(name = "tarifa_consulta", nullable = false, precision = 10, scale = 2)
    var tarifaConsulta: BigDecimal,

    @OneToOne
    @JoinColumn(name = "id_usuario", nullable = false)
    var usuario: Usuario,

    @Column(precision = 12, scale = 2)
    var sueldo: BigDecimal? = null,

    @Column(precision = 12, scale = 2)
    var bono: BigDecimal? = null
)
